modules/articles/client/config/articles.client.routes.js

```diff
@@ -12,25 +16,59 @@ angular.module('articles').config(['$stateProvider',
       })
       .state('articles.list', {
         url: '',
-        templateUrl: 'modules/articles/client/views/list-articles.client.view.html'
+        templateUrl: 'modules/articles/client/views/list-articles.client.view.html',
+        controller: 'ArticlesListController',
+        controllerAs: 'vm'
       })
       .state('articles.create', {
         url: '/create',
-        templateUrl: 'modules/articles/client/views/create-article.client.view.html',
+        templateUrl: 'modules/articles/client/views/form-article.client.view.html',
+        controller: 'ArticlesController',
+        controllerAs: 'vm',
+        resolve: {
+          articleResolve: newArticle
+        },
         data: {
           roles: ['user', 'admin']
+        },
+        resolve: {
+          articleResolve: newArticle
         }
       })
-      .state('articles.view', {
-        url: '/:articleId',
-        templateUrl: 'modules/articles/client/views/view-article.client.view.html'
-      })
       .state('articles.edit', {
         url: '/:articleId/edit',
-        templateUrl: 'modules/articles/client/views/edit-article.client.view.html',
+        templateUrl: 'modules/articles/client/views/form-article.client.view.html',
+        controller: 'ArticlesController',
+        controllerAs: 'vm',
+        resolve: {
+          articleResolve: getArticle
+        },
         data: {
           roles: ['user', 'admin']
         }
+      })
+      .state('articles.view', {
+        url: '/:articleId',
+        templateUrl: 'modules/articles/client/views/view-article.client.view.html',
+        controller: 'ArticlesController',
+        controllerAs: 'vm',
+        resolve: {
+          articleResolve: getArticle
+        }
       });
   }
-]);
+
+  getArticle.$inject = ['$stateParams', 'Article'];
+
+  function getArticle($stateParams, Article) {
+    return Article.get({
+      articleId: $stateParams.articleId
+    }).$promise;
+  }
+
+  newArticle.$inject = ['Article'];
+
+  function newArticle(Article) {
```

Comment:
@ilanbiala It makes the controller smaller and allows the use of UI-Router nested views to share the resolve between the views and other controllers.